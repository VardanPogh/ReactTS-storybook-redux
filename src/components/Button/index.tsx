import React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import clsx from 'clsx'

const useStyles = makeStyles(() =>
  createStyles({
    contained: {
      minWidth: 105,
      minHeight: 55,
      backgroundColor: '#014bad',
      color: '#fff',
      '&:hover': {
        backgroundColor: '#fff',
        color: '#014bad',
      },
    },
    asLink: {
      minWidth: 105,
      minHeight: 55,
      color: '#474747',
      '&:hover': {
        textDecoration: 'underline',
      },
    },
    outlined: {
      minWidth: 105,
      minHeight: 55,
      color: '#014bad',
      border: '1px solid #014bad',
    },
  })
)

interface Props {
  variant?: string
  text: string
  onClick?(): void
  className?: string
  type?: 'button' | 'reset' | 'submit'
}

const ButtonStyled: React.FC<Props> = ({
  variant,
  text,
  onClick,
  className,
  type,
}) => {
  const classes = useStyles()
  switch (variant) {
    case 'contained':
      return (
        <Button
          type={type}
          onClick={onClick}
          variant="contained"
          className={clsx(classes.contained, className)}
        >
          {text}
        </Button>
      )
    case 'outlined':
      return (
        <Button
          type={type}
          onClick={onClick}
          variant="outlined"
          className={clsx(classes.outlined, className)}
        >
          {text}
        </Button>
      )

    default:
      return (
        <Button
          type={type}
          onClick={onClick}
          className={clsx(classes.asLink, className)}
        >
          {text}
        </Button>
      )
  }
}

export default ButtonStyled
