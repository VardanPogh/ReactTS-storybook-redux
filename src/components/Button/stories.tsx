import React from 'react'
import { withKnobs, text, select } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import Button from '.'

const options = {
  Outlined: 'outlined',
  Contained: 'contained',
  None: '',
}

export default {
  title: 'Button',
  component: Button,
  decorators: [withKnobs],
}

export const Default: React.FC = () => (
  <Button
    onClick={action('button-click')}
    variant={select('Variant', options, options.Contained)}
    text={text('Label', 'Login')}
  />
)
