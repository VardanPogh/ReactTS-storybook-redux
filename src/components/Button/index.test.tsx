import React from 'react'
import { render } from '@testing-library/react'

import Button from '.'

describe('Button Test', () => {
  test('Load Button', () => {
    const myMock = jest.fn()
    const { container } = render(
      <Button variant="contained" text="submit" onClick={myMock} />
    )
    expect(container).toMatchSnapshot()
  })
})
