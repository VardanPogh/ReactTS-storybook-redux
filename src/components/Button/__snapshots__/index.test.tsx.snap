// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test Load Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
